src/app/tests/TestBindingTable.cpp

```diff
@@ -15,122 +15,122 @@
  *    limitations under the License.
  */
 
+#include <pw_unit_test/framework.h>
+
 #include <app/util/binding-table.h>
 #include <app/util/config.h>
 #include <lib/support/DefaultStorageKeyAllocator.h>
 #include <lib/support/TestPersistentStorageDelegate.h>
 #include <lib/support/UnitTestRegistration.h>
-#include <nlunit-test.h>
 
 using chip::BindingTable;
 
 namespace {
 
-void TestEmptyBindingTable(nlTestSuite * aSuite, void * aContext)
+void VerifyTableSame(BindingTable & table, const std::vector<EmberBindingTableEntry> & expected)
+{
+    EXPECT_EQ(table.Size(), expected.size());
+    auto iter1 = table.begin();
+    auto iter2 = expected.begin();
+    while (iter2 != expected.end())
+    {
+        EXPECT_NE(iter1, table.end());
```

Comment:
if we use EXPECT then processing continues and the line below will deference invalid data probably.

Should we maybe do ASSERT_EQ on line 32 above and then not need this?

Reply:
yes sure, assuming that iter1 will not have an invalid index?